src/Commands/CommandGroup.cs

```diff
@@ -0,0 +1,209 @@
+
+            if (!result.IsSuccess)
+            {
+                await feedbackService.SendContextualErrorAsync($"An error occurred:\n\n{result.Error}");
+                return result;
+            }
+
+            return result;
+        }
+        catch (Exception ex)
+        {
+            await feedbackService.SendContextualErrorAsync($"An error occurred:\n\n{ex}");
+            return (Result)new UnhandledExceptionError(ex);
+        }
+    }
+
+
+    [Command("portal")]
+    public async Task<IResult> PortalAsync(string destinationChannelName)
+    {
+        if (!context.TryGetChannelID(out var sourceChannelId))
+            return Result.FromError(new InvalidOperationError("Could not determine the source channel."));
+
+        var sourceChannelResult = await channelApi.GetChannelAsync(sourceChannelId);
+        if (!sourceChannelResult.IsSuccess)
+            return Result.FromError(sourceChannelResult.Error);
+
+        if (sourceChannelResult.Entity.GuildID == null)
+            return Result.FromError(new NotFoundError("This command can only be used inside a server."));
+
+        var guildId = sourceChannelResult.Entity.GuildID.Value;
+
+        var channelsResult = await guildApi.GetGuildChannelsAsync(guildId);
+        if (!channelsResult.IsSuccess)
+            return Result.FromError(channelsResult.Error);
+
+        var destinationChannel = channelsResult.Entity
+            .FirstOrDefault(c => string.Equals(c.Name.Value, destinationChannelName, StringComparison.OrdinalIgnoreCase));
+
+        if (destinationChannel == null)
+            return Result.FromError(new NotFoundError($"Channel '{destinationChannelName}' not found."));
+
+        var destinationChannelId = destinationChannel.ID;
+
+
+        if (!context.TryGetUserID(out var userId))
+            return Result.FromError(new InvalidOperationError("Could not determine the user ID."));
+
+        var botPermissionsResult = await guildApi.GetGuildMemberAsync(guildId, userId);
+        if (!botPermissionsResult.IsSuccess)
+            return Result.FromError(botPermissionsResult.Error);
+
+
+        var botPermissions = botPermissionsResult.Entity.Roles
+            .Select(roleId => guildApi.GetGuildRolesAsync(guildId, CancellationToken.None).Result.Entity.First(role => role.ID == roleId).Permissions);
+
+
+        DiscordPermission permissions = default;
+
+        foreach (var set in botPermissions)
+        {
+            if (set.HasPermission(DiscordPermission.SendMessages))
+            {
+                permissions = DiscordPermission.SendMessages;
+                break;
+            }
+        }
+
+        if (permissions == default)
+            return Result.FromError(new InvalidOperationError("Bot does not have permission to send messages in the destination channel."));
+
+        var sourceChannelMention = $"<#{sourceChannelId}>";
+        var destinationChannelMention = $"<#{destinationChannelId}>";
+
+        var outPortalEmbed = new Embed(
```

Comment:
The code from here on needs to be rearranged so that the portals link to the portal messages, not just the channel it was opened in. See 5-7 in the "Detailed specs" section of #7.